pytorch_lightning/trainer/training_loop.py

```diff
@@ -834,9 +835,9 @@ def should_check_val_fx(self, batch_idx, is_last_batch, on_epoch=False):
 
         return should_check_val and can_check_val
 
-    def build_train_args(self, batch, batch_idx, opt_idx, hiddens):
+    def _build_kwargs(self, batch, batch_idx, opt_idx, hiddens):
         # enable not needing to add opt_idx to training_step
-        args = [batch, batch_idx]
+        kwargs = OrderedDict({'batch': batch, 'batch_idx': batch_idx})
```

Comment:
This does not work as the moment you build the inner dict you lose the ordered guarantee. You need to use a list of tuples

Reply:
done.